scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -605,6 +613,49 @@ s32 BPF_STRUCT_OPS(rusty_select_cpu, struct task_struct *p, s32 prev_cpu,
 			}
 		}
 	}
+	
+	// the task might also be forked/execd so check for wake up only
+	if (wake_flags & SCX_WAKE_TTWU) {
+		cpu = scx_bpf_pick_idle_cpu((const struct cpumask *)p_cpumask, 0);
+		if (cpu >= 0) {
+			goto direct;
+		}
+		if (scx_bpf_dsq_nr_queued(taskc->dom_id) == 0) {
+			s32 lowestWeight = -1;
+			s32 foundCPU = -1;
+			u32 currentWeight = taskc->weight;
+			u64 currentSumRunTime = p->se.sum_exec_runtime;
+			struct cpu_task_data *ctd;
+			// Look for the currently running task with the lowest weight
+			// and with a weight lower than the current task
+			// and with a higher sum_exec_runtime than the current task
+			bpf_for(cpu, 0, nr_cpus) {
+				if (cpu_to_dom_id(cpu) != taskc->dom_id) {
+					continue;
+				}
+				ctd = MEMBER_VPTR(cpu_task_data, [cpu]);
+				if (!ctd) {
+					continue;
+				}
+				if (!ctd->has_running_task) {
+					// No running task, let's just use this cpu
+					foundCPU = cpu;
+					break;
+				} else if (ctd->weight < currentWeight &&
+					currentSumRunTime < ctd->sum_exec_runtime &&
```

Comment:
I'm not sure this makes sense. Why are we comparing the tasks' accumulated runtimes? Shouldn't we be comparing vtimes instead?

Reply:
@Decave Mentioned something about fairness e.g. this is meant to protect a lower priority task always getting kicked off the cpu by a higher priority task. vtime is a combination of accumulated runtimes and weight right? So should I just be using that instead of these separately?

Reply:
Total accumulated execution time doesn't have much to do with fairness unless you're comparing tasks which are started at the same time across their entire lifetimes. vtime is what signifies their current relative priority.